rsp_vision/save/tables_manager.py

```diff
@@ -0,0 +1,52 @@
+from pathlib import Path
+
+import pandas as pd
+
+
+class AnalysisSuccessTable:
+    def __init__(self, path: Path):
+        self.path = path / "analysis_success.log"
+
+        if not self.path.exists():
+            self.df = pd.DataFrame(
+                columns=["dataset_name", "date", "latest_job_id", "error"]
+            )
+            self.df.to_csv(self.path)
+        else:
+            self.df = pd.read_csv(self.path, index_col=0, header=0)
+
+    def update(
+        self, dataset_name: str, date: str, latest_job_id: int, error: str
+    ):
+        if not self.is_dataset_in_table(dataset_name):
```

Comment:
could swap the order of the conditions and change the name of the check for readabiltiy (e.g.  `if self.dataset_is_in_table: ... else: ...`